### PR TITLE
Add `list()` for retrieving registered fail points

### PR DESCRIPTION
**Pull request by overvenus**
PTAL @BusyJay @siddontang

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -225,14 +225,16 @@ struct FailPoint {
     pause: Mutex<bool>,
     pause_notifier: Condvar,
     actions: RwLock<Vec<Action>>,
+    terms: String,
```
Comment (BusyJay) [inline src/lib.rs]: What does `term` mean?

File: src/lib.rs [hunk 2]
```diff
@@ -349,6 +351,17 @@ pub fn teardown() {
...
 }
 
+/// Get all registered fail points.
+///
+/// Return a vector of `(name, actions)` pairs.
+pub fn list() -> Vec<(String, String)> {
+    let registry = REGISTRY.registry.read().unwrap();
+    registry
+        .iter()
+        .map(|(k, v)| (k.to_string(), v.terms.clone()))
```
Comment (BusyJay) [inline src/lib.rs]: I think you should list actions instead of `terms`.

File: tests/tests.rs [hunk 3]
```diff
@@ -189,3 +189,9 @@ fn test_condition() {
 
     assert_eq!(f(true), 2);
 }
+
+#[test]
+fn test_list() {
+    fail::cfg("tests::list", "off").unwrap();
+    assert!(fail::list().contains(&("tests::list".to_string(), "off".to_string())));
```
Comment (BusyJay) [inline tests/tests.rs]: What if the failpoint exists already?

File: src/lib.rs [hunk 4]
```diff
@@ -344,11 +343,22 @@ pub fn teardown() {
...
 }
 
+/// Get all registered fail points.
+///
+/// Return a vector of `(name, actions)` pairs.
+pub fn list() -> Vec<(String, String)> {
+    let registry = REGISTRY.registry.read().unwrap();
+    registry
+        .iter()
+        .map(|(k, v)| (k.to_string(), v.actions.read().unwrap().1.clone()))
```
Comment (BusyJay) [inline src/lib.rs]: If a failpoint is paused, it will block till a new action is set.

**Comment by BusyJay** [comment]
LGTM

File: src/lib.rs [hunk 5]
```diff
@@ -225,25 +225,27 @@ struct FailPoint {
...
             pause_notifier: Condvar::new(),
-            actions: RwLock::new(vec![]),
+            actions: RwLock::default(),
+            actions_str: RwLock::default(),
         }
     }
 
-    #[allow(dead_code)]
-    fn set_actions(&self, actions: Vec<Action>) {
+    fn set_actions(&self, actions: Vec<Action>, actions_str: &str) {
```
Comment (siddontang) [inline src/lib.rs]: I guess `set_actions(name, acrtions)` may be more reasonable.

**Pull request merged**